File: app/src/main/java/wjy/yo/ereader/entity/book/Para.java
package wjy.yo.ereader.entity.book;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Ignore;
import android.support.annotation.NonNull;

import wjy.yo.ereader.entity.FetchedData;
import wjy.yo.ereader.entity.Ordered;
import wjy.yo.ereader.vo.WordContext;

@Entity(tableName = "book_para")
public class Para extends FetchedData implements Ordered {

    @NonNull
    @ForeignKey(entity = Book.class, parentColumns = "id", childColumns = "bookId")
    private String bookId;

    @NonNull
    @ForeignKey(entity = Chap.class, parentColumns = "id", childColumns = "chapId")
    private String chapId;

    private String content;

    private String trans;

    private long no;

    @Ignore
    private int seq;

    @NonNull
    public String getBookId() {
        return bookId;
    }

    public void setBookId(@NonNull String bookId) {
        this.bookId = bookId;
    }

    @NonNull
    public String getChapId() {
        return chapId;
    }

    public void setChapId(@NonNull String chapId) {
        this.chapId = chapId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getTrans() {
        return trans;
    }

    public void setTrans(String trans) {
        this.trans = trans;
    }

    public long getNo() {
        return no;
    }

    public void setNo(long no) {
        this.no = no;
    }

    public int getSeq() {
        return seq;
    }

    public void setSeq(int seq) {
        this.seq = seq;
    }

    public WordContext getWordContext() {
        WordContext wc = new WordContext();
        wc.setParaId(this.id);
        wc.setChapId(this.chapId);
        wc.setBookId(this.bookId);
        return wc;
    }
}

File: app/src/main/java/wjy/yo/ereader/entity/book/Para.kt
package wjy.yo.ereader.entity.book

import android.arch.persistence.room.Entity
import android.arch.persistence.room.ForeignKey

import wjy.yo.ereader.entity.FetchedData
import wjy.yo.ereader.entity.Ordered

@Entity(tableName = "book_para")
class Para : FetchedData(""), Ordered {

    @ForeignKey(entity = Book::class, parentColumns = arrayOf("id"), childColumns = arrayOf("bookId"))
    var bookId: String? = null

    @ForeignKey(entity = Chap::class, parentColumns = arrayOf("id"), childColumns = arrayOf("chapId"))
    var chapId: String? = null

    var content: String? = null

    var trans: String? = null

    override var no: Long = 0
}
